### ISSUE TITLE
[v4 Training] Issue 1: Stage 1 StabilityLossの勾配消失

### ISSUE THREAD

**Issue by tamatebox** [tamatebox/samadhi_model#2]
## 症状
- Stage 1のStabilityLossが約1.023から下がらない
- 学習が進んでも収束しない

## 原因
**SantanaLogが状態を`.detach()`して保存していた**

```python
def add(self, state: torch.Tensor, ...):
    self.states.append(state.detach().cpu())  # ← 勾配が切れる
```

StabilityLossは `||s_T - s_{T-1}||²` を計算するが、SantanaLogから取得した状態には勾配がないため、backpropagationが不可能だった。

## 対処
`stability_pair`を導入し、SantanaLogとは別に勾配付きの状態ペアを保持

```python
class SamathaOutput(NamedTuple):
    s_star: torch.Tensor
    santana: SantanaLog
    severity: torch.Tensor
    stability_pair: Tuple[torch.Tensor, torch.Tensor]  # (s_T, s_{T-1}) with gradients
```

## 教訓
- ログ用データと学習用データは分離する
- `.detach()`が必要な理由を明示的にコメントで残す

## 詳細
[docs/issues/001_training_issues_v4.md](https://github.com/tamatebox/samadhi_model/blob/main/docs/issues/001_training_issues_v4.md#issue-1-stage-1-stabilitylossの勾配消失)

## 関連ファイル
- `satipatthana/core/engines.py`
- `satipatthana/core/santana.py`
